apps/prairielearn/elements/pl-number-input/pl-number-input.py

```diff
@@ -65,47 +86,71 @@ def prepare(element_html, data):
+) -> str:
     a_tru = pl.from_json(data["correct_answers"].get(name, None))
     if a_tru is not None:
         # Get format and comparison parameters
         custom_format = pl.get_string_attrib(element, "custom-format", None)
-        comparison = pl.get_string_attrib(element, "comparison", COMPARISON_DEFAULT)
+        comparison = pl.get_enum_attrib(
+            element, "comparison", ComparisonType, COMPARISON_DEFAULT
+        )
+
         if custom_format is not None:
             a_tru = ("{:" + custom_format + "}").format(a_tru)
-        elif comparison == "relabs":
+        elif comparison is ComparisonType.RELABS:
             # FIXME: render correctly with respect to rtol and atol
             a_tru = "{:.12g}".format(a_tru)
-        elif comparison == "sigfig":
+        elif comparison is ComparisonType.SIGFIG:
             digits = pl.get_integer_attrib(element, "digits", DIGITS_DEFAULT)
             a_tru = pl.string_from_number_sigfig(a_tru, digits=digits)
-        elif comparison == "decdig":
+        elif comparison is ComparisonType.DECDIG:
             digits = pl.get_integer_attrib(element, "digits", DIGITS_DEFAULT)
             a_tru = "{:.{ndigits}f}".format(a_tru, ndigits=digits)
         else:
-            raise ValueError(
-                'method of comparison "%s" is not valid (must be "relabs", "sigfig", or "decdig")'
-                % comparison
-            )
+            assert_never(comparison)
     return a_tru
 
 
-def render(element_html, data):
+def get_string_precision(number_string: str) -> float:
+    if "." in number_string:
+        return 10 ** -len(number_string.partition(".")[2])
+    else:
+        return 10 ** (len(number_string) - len(number_string.rstrip("0")))
+
+
+def get_string_significant_digits(number_string: str) -> int:
+    if "." in number_string:
+        number_string_partition = number_string.partition(".")
+        integer_part = len(number_string_partition[0])
+        decimal_part = len(number_string_partition[2].lstrip("0"))
+        return integer_part + decimal_part
+    else:
+        return len(number_string.rstrip("0"))
+
+
+def render(element_html: str, data: pl.QuestionData) -> str:
     element = lxml.html.fragment_fromstring(element_html)
     name = pl.get_string_attrib(element, "answers-name")
     label = pl.get_string_attrib(element, "label", None)
     suffix = pl.get_string_attrib(element, "suffix", None)
-    display = pl.get_string_attrib(element, "display", DISPLAY_DEFAULT)
+    display = pl.get_enum_attrib(element, "display", DisplayType, DISPLAY_DEFAULT)
     allow_fractions = pl.get_boolean_attrib(
         element, "allow-fractions", ALLOW_FRACTIONS_DEFAULT
     )
     custom_format = pl.get_string_attrib(
         element, "custom-format", CUSTOM_FORMAT_DEFAULT
     )
+    show_score = pl.get_boolean_attrib(element, "show-score", SHOW_SCORE_DEFAULT)
+    accessibility_description = pl.get_string_attrib(
```

Comment:
I think this can be removed, since the custom description is just being run through a hidden label (but we have the `aria-labeledby` set independently).

Reply:
As I remember, these were added per #7385, but I think we could remove these for now and I can open another PR to address the accessibility issues later, since you mentioned it would be a good idea to get what we have here merged first before the flood gates of September opens. 

@Frozemint note to self: open the accessibility PR